presto-spi/src/main/java/com/facebook/presto/spi/statistics/ColumnStatistics.java

```diff
@@ -11,56 +11,52 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package com.facebook.presto.spi.statistics;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
 
 import static com.facebook.presto.spi.statistics.Estimate.unknownValue;
-import static java.util.Collections.unmodifiableMap;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.unmodifiableList;
 import static java.util.Objects.requireNonNull;
 
 public final class ColumnStatistics
 {
-    private final Map<String, Estimate> statistics;
-    private static final String DATA_SIZE_STATISTIC_KEY = "data_size";
-    private static final String NULLS_FRACTION_STATISTIC_KEY = "nulls_fraction";
-    private static final String DISTINCT_VALUES_STATITIC_KEY = "distinct_values_count";
+    private static final List<RangeColumnStatistics> SINGLE_UNKNOWN_RANGE_STATISTICS = singletonList(RangeColumnStatistics.builder().build());
 
-    private ColumnStatistics(Estimate dataSize, Estimate nullsCount, Estimate distinctValuesCount)
-    {
-        requireNonNull(dataSize, "dataSize can not be null");
-        statistics = createStatisticsMap(dataSize, nullsCount, distinctValuesCount);
-    }
+    private final List<RangeColumnStatistics> rangeColumnStatistics;
+    private final Estimate nullsFraction;
```

Comment:
Maybe it needs to remain a `Map`. Why was `Map` was used to keep statistics previously? Does it matter do you have tree or single entry?